PurchasesCoreSwift/Public/EntitlementInfo.swift

```diff
@@ -0,0 +1,346 @@
+//
+//  EntitlementInfo.swift
+//  PurchasesCoreSwift
+//
+//  Created by Joshua Liebowitz on 6/25/21.
+//  Copyright © 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+
+/**
+ Enum of supported stores
+ */
+@objc(RCStore) public enum Store: Int {
+    /// For entitlements granted via Apple App Store.
+    @objc(RCAppStore) case appStore = 0
+    /// For entitlements granted via Apple Mac App Store.
+    @objc(RCMacAppStore) case macAppStore = 1
+    /// For entitlements granted via Google Play Store.
+    @objc(RCPlayStore) case playStore = 2
+    /// For entitlements granted via Stripe.
+    @objc(RCStripe) case stripe = 3
+    /// For entitlements granted via a promo in RevenueCat.
+    @objc(RCPromotional) case promotional = 4
+    /// For entitlements granted via an unknown store.
+    @objc(RCUnknownStore) case unknownStore = 5
+}
+
+/**
+ Enum of supported period types for an entitlement.
+ */
+@objc(RCPeriodType) public enum PeriodType: Int {
+    /// If the entitlement is not under an introductory or trial period.
+    @objc(RCNormal) case normal = 0
+    /// If the entitlement is under a introductory price period.
+    @objc(RCIntro) case intro = 1
+    /// If the entitlement is under a trial period.
+    @objc(RCTrial) case trial = 2
+}
+
+@objcMembers @objc(RCEntitlementInfo) public class EntitlementInfo: NSObject {
+    /**
+     The entitlement identifier configured in the RevenueCat dashboard
+     */
+    public let identifier: String
+
+    /**
+     True if the user has access to this entitlement
+     */
+    public let isActive: Bool
+
+    /**
+     True if the underlying subscription is set to renew at the end of
+     the billing period (expirationDate). Will always be True if entitlement
+     is for lifetime access.
+     */
+    public let willRenew: Bool
+
+    /**
+     The last period type this entitlement was in
+     Either: RCNormal, RCIntro, RCTrial
+     */
+    public let periodType: PeriodType
+
+    /**
+     The latest purchase or renewal date for the entitlement.
+     */
+    public let latestPurchaseDate: Date? // TODO: This used to be NON-NULL
+
+    /**
+     The first date this entitlement was purchased
+     */
+    public let originalPurchaseDate: Date? // TODO: This used to be NON-NULL
+
+    /**
+     The expiration date for the entitlement, can be `nil` for lifetime access.
+     If the `periodType` is `trial`, this is the trial expiration date.
+     */
+    public private(set) var expirationDate: Date?
```

Comment:
`expirationDate` is only set at `init`, right? I don't think this needs to be `var`

Reply:
same for `unsubscribeDetectedAt` and `billingIssueDetectedAt`

Reply:
✅